plugins/CoreAdminHome/vue/src/JsTrackingCodeGenerator/JsTrackingCodeGenerator.vue

```diff
@@ -63,7 +63,10 @@
         </div>
         </div>
         <div id="javascript-text">
-          <pre v-select-on-focus="{}" class="codeblock" v-text="trackingCode" ref="trackingCode"/>
+          <div>
```

Comment:
That was the approach I took initially, but changing the parent element of vue components broke the  bindings and css in many cases, so I went for a less intrusive approach. It's only a simple classless div which is required and this also allows the option to have the button above the code block rather than overlaying it if required.

Reply:
Ok. I see. Guess it's fine to keep it that way then.

Reply:
I also feel like having the extra div everywhere is prone to errors over time, more so if ~80% already have some wrapping div around the <pre><code> block.
First I was surprised the original mechanism was just a directive, now I'd say this might actually be a simple component, single or multiline, that would take care of all it needs within itself, externally as a single tag, and only receive the code as its content via a prop.
I think it will also better support use cases such as if we wanted to do dynamic code updates in future based on config options without having to do bigger reloads.
Just my 2 cents.

Reply:
I was also wondering if a component would be a better fit. But thought that this would restrict some possibilities like being able to bind event handlers on the code block or similar. But maybe that would work out of the box when using slots 🤔

Reply:
I don't think the wrapping div is such a big problem,. Sure a component might have been nicer, but we have a working change which has now had time invested in it, requires minimal rework of existing templates and there are other parts of the application which are more deserving of improvement. Perfect is the enemy of good enough :slightly_smiling_face:

Reply:
Fair enough. If we come across similar stuff in the future, it might be a good candidate for an approach discussion before implementation.

Reply:
Definitely :+1:  
We're still in the early stages with vue, there has been a direct one-to-one conversion of all the old angular code into vue components, but now there is plenty of discussion to be had on what approaches we take when implementing new functionality and when to rework existing code (or not)